pages/edit.js

```diff
@@ -0,0 +1,365 @@
+import Image from 'next/image';
+import styled from 'styled-components';
+import hook from '../public/hook.svg';
+import add from '../public/add-button.svg';
+import remove from '../public/delete-button.svg';
+import FormHeadline from '../components/EditFormComponents/FormHeadline';
+import EditBackgroundImage from '../components/EditFormComponents/EditBackgroundImage';
+import useLocalStorage from '../hooks/useLocalStorage';
+import { useRouter } from 'next/router';
+
+export function changeTimerChain(event, id, newTimerChain, setNewTimerChain) {
+  if (event.target.id === 'name') {
+    setNewTimerChain(
+      newTimerChain.map((newTimer) => {
+        if (newTimer.id === id) {
+          return {
+            ...newTimer,
+            name: event.target.value,
+          };
+        }
+        return newTimer;
+      })
+    );
+  } else if (event.target.id === 'seconds') {
+    setNewTimerChain(
+      newTimerChain.map((newTimer) => {
+        if (newTimer.id === id) {
+          return { ...newTimer, seconds: Number(event.target.value) };
+        }
+        return newTimer;
+      })
+    );
+  } else {
+    setNewTimerChain(
+      newTimerChain.map((newTimer) => {
+        if (newTimer.id === id) {
+          return { ...newTimer, minutes: Number(event.target.value) };
+        }
+        return newTimer;
+      })
+    );
+  }
+}
+
+export function handleAdd(newTimerChain, setNewTimerChain) {
+  const addId = newTimerChain.length + 1;
+  setNewTimerChain([
+    ...newTimerChain,
+    { id: addId, minutes: 5, seconds: 0, name: '' },
+  ]);
+}
+
+export function handleRemove(newTimerChain, setNewTimerChain) {
+  const lastId = newTimerChain.length;
+  if (lastId > 1) {
+    const newChain = newTimerChain.filter((timer) => timer.id !== lastId);
+    setNewTimerChain(newChain);
```

Comment:
If I remember correctly your remove button removes the last timer in the chain. Instead of filter you could work with the array method .pop() which removes the last entry of the array. 
This way you will always remove the entry that has the current highest ID.